libs/server/Storage/Session/ObjectStore/HashOps.cs

```diff
@@ -537,5 +538,98 @@ public GarnetStatus HashIncrement<TObjectContext>(byte[] key, ArgSlice input, ou
+        /// <param name="isMilliseconds">Indicates whether the TTL is in milliseconds.</param>
+        /// <param name="isTimestamp">Indicates whether the TTL is a timestamp.</param>
+        /// <param name="input">The input object containing the operation details.</param>
+        /// <param name="outputFooter">The output footer object to store the result.</param>
+        /// <param name="objectContext">The object context for the operation.</param>
+        /// <returns>The status of the operation.</returns>
+        public GarnetStatus HashTimeToLive<TObjectContext>(ArgSlice key, bool isMilliseconds, bool isTimestamp, ref ObjectInput input, ref GarnetObjectStoreOutput outputFooter, ref TObjectContext objectContext)
+            where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+        {
+            var innerInput = new ObjectInput(input.header, ref input.parseState, arg1: isMilliseconds ? 1 : 0, arg2: isTimestamp ? 1 : 0);
+
+            return ReadObjectStoreOperationWithOutput(key.ToArray(), ref innerInput, ref objectContext, ref outputFooter);
+        }
+
+        /// <summary>
+        /// Removes the expiration time from a hash key, making it persistent.
+        /// </summary>
+        /// <typeparam name="TObjectContext">The type of the object context.</typeparam>
+        /// <param name="key">The key of the hash.</param>
+        /// <param name="input">The input object containing the operation details.</param>
+        /// <param name="outputFooter">The output footer object to store the result.</param>
+        /// <param name="objectContext">The object context for the operation.</param>
+        /// <returns>The status of the operation.</returns>
+        public GarnetStatus HashPersist<TObjectContext>(ArgSlice key, ref ObjectInput input, ref GarnetObjectStoreOutput outputFooter, ref TObjectContext objectContext)
+            where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+            => RMWObjectStoreOperationWithOutput(key.ToArray(), ref input, ref objectContext, ref outputFooter);
+
+        public GarnetStatus HashCollect<TObjectContext>(ArgSlice key, ref ObjectInput input, ref TObjectContext objectContext)
+            where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+        {
+            if (key.ReadOnlySpan.SequenceEqual("*"u8))
+            {
+                long cursor = 0;
+                long storeCursor = 0;
+
+                // Scan all hash keys in batches
+                do
+                {
+                    if (!DbScan(key, true, cursor, out storeCursor, out var hashKeys, 100, CmdStrings.HASH))
```

Comment:
I would assume this is highly inefficient to find all the Hash objects and then call HCOLLECT on them. Do you guys have any idea to find only the hash object that has expiring fields?

Reply:
Yes, I agree. A possible solution will be to store the key that have expirations in the store itself as a SetObject under a predetermined key (e.g. __keysWithExpiration__), so you can access them really easily and fast.

Reply:
closing because it seems good enough for now.